### PR TITLE
IntelFsp2WrapperPkg/FspmWrapperPeim: Fix FspT/M address for measurement

### PR DESCRIPTION
**Pull request by zhihaoli1064**
REF: https://bugzilla.tianocore.org/show_bug.cgi?id=4716

TCG module should use permanent address of FSP-T/M for
measurement. TCG notification checks MigatedFvInfoHob
and transmits DRAM address for measurement

Cc: Chasel Chiu <chasel.chiu@intel.com>
Cc: Nate DeSimone <nathaniel.l.desimone@intel.com>
Cc: Duggapu Chinni B <chinni.b.duggapu@intel.com>
Cc: Chen Gang C <gang.c.chen@intel.com>
Cc: Liming Gao <gaoliming@byosoft.com.cn>

### CODE REVIEW

File: IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c [hunk 1]
```diff
@@ -278,27 +279,50 @@ TcgPpiNotify (
...
   DEBUG ((DEBUG_INFO, "TcgPpiNotify FSPM\n"));
 
-  FspMeasureMask = PcdGet32 (PcdFspMeasurementConfig);
+  FspMeasureMask  = PcdGet32 (PcdFspMeasurementConfig);
+  FsptBaseAddress = (EFI_PHYSICAL_ADDRESS)PcdGet32 (PcdFsptBaseAddress);
+  FspmBaseAddress = (EFI_PHYSICAL_ADDRESS)PcdGet32 (PcdFspmBaseAddress);
+  Hob.Raw         = GetFirstGuidHob (&gEdkiiMigratedFvInfoGuid);
+  while (Hob.Raw != NULL) {
+    MigratedFvInfo = GET_GUID_HOB_DATA (Hob);
+    if ((MigratedFvInfo->FvOrgBase == (UINT32)(UINTN)PcdGet32 (PcdFsptBaseAddress)) && (MigratedFvInfo->FvDataBase != 0)) {
```
Comment (tedkuo1) [inline IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c:296]: Why do we need type casting (UINT32)(UINTN) in this line?

File: IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c [hunk 2]
```diff
@@ -278,27 +279,50 @@ TcgPpiNotify (
...
+      //
+      // Found the migrated FspT raw data
+      //
+      FsptBaseAddress = MigratedFvInfo->FvDataBase;
+    }
+
+    if ((MigratedFvInfo->FvOrgBase == (UINT32)(UINTN)PcdGet32 (PcdFspmBaseAddress)) && (MigratedFvInfo->FvDataBase != 0)) {
```
Comment (tedkuo1) [inline IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c:303]: Why do we need type casting (UINT32)(UINTN) in this line?

File: IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c (hunk 1 above)
Comment (zhihaoli1064) [reply IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c:296]: I think we don't need it here. I will change it.

File: IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c (hunk 2 above)
Comment (zhihaoli1064) [reply IntelFsp2WrapperPkg/FspmWrapperPeim/FspmWrapperPeim.c:303]: same as the above

**Review by tedkuo1 — APPROVED** [review]

**Pull request merged**